src/Nox.Types/Types/AutoNumber/AutoNumber.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace Nox.Types;
+
+/// <summary>
+/// Represents a Nox <see cref="AutoNumber"/> type and value object.
+/// </summary>
+public sealed class AutoNumber : ValueObject<uint, AutoNumber>
+{
+    public AutoNumber()
+    {
+        Value = 0;
+    }
+
+    /// <summary>
+    /// Creates a <see cref="AutoNumber"/> object with the initial value of <paramref name="value"/>.
+    /// </summary>
+    /// <param name="value">Value to be parsed.</param>
+    /// <returns>New <see cref="AutoNumber"/> object.</returns>
+    public new static AutoNumber From(uint value) => new() { Value = value };
```

Comment:
Autonumber cannot never be created, the From should raise an exception

Should only Load from the Database

Reply:
`From` method now raises an exception.